### ISSUE TITLE
No Json serializer found for type error && not enough arguments for method toJson error

### ISSUE THREAD

**Issue by wontheone1** [unicredit/sbt-swagger-codegen-examples#1]
Hello, I am trying to test this codegen example but maybe I am missing something, but I find it hard to follow it straight forwardly, and suspecting maybe this example is not self sufficient example.

What I did was, 1. clone example, 2. build it using 'sbt', 3. run 'swaggerCodeGen, swaggerPlayClientCodeGen, swaggerPlayServerCodeGen' commands in the order, also uncomment in build.sbt the dependency from play-ws. 4. uncomment PetStoreControllerImpl file(The part where it says 
// use this code in case of :
// swaggerServerAsync := true)
5. I refactored PetStoreControllerImpl.scala file to same location as where Model.scala is because, in my intelli J the model was not imported properly and giving me red colors. 
6. Try to build it. Then I get 8 errors in PetStoreController.scala file, which I wrote in the title.
Am I doing something wrong? or could you fix this problem?
# Thank you very much. (below are my error messages)

Error:(12, 21) No Json serializer found for type scala.concurrent.Future[scala.collection.mutable.Seq[swagger.codegen.pet]]. Try to implement an implicit Writes or Format for this type.
      Ok(Json.toJson(findPetsImpl(limit, tags)))
                    ^
Error:(12, 21) not enough arguments for method toJson: (implicit tjs: play.api.libs.json.Writes[scala.concurrent.Future[scala.collection.mutable.Seq[swagger.codegen.pet]]])play.api.libs.json.JsValue.
Unspecified value parameter tjs.
      Ok(Json.toJson(findPetsImpl(limit, tags)))

Error:(22, 21) No Json serializer found for type scala.concurrent.Future[swagger.codegen.newPet]. Try to implement an implicit Writes or Format for this type.
      Ok(Json.toJson(addPetImpl(pet)))
                    ^
Error:(22, 21) not enough arguments for method toJson: (implicit tjs: play.api.libs.json.Writes[scala.concurrent.Future[swagger.codegen.newPet]])play.api.libs.json.JsValue.
Unspecified value parameter tjs.
      Ok(Json.toJson(addPetImpl(pet)))
                    ^
Error:(31, 21) No Json serializer found for type scala.concurrent.Future[swagger.codegen.pet]. Try to implement an implicit Writes or Format for this type.
      Ok(Json.toJson(deletePetImpl(id)))
                    ^
Error:(31, 21) not enough arguments for method toJson: (implicit tjs: play.api.libs.json.Writes[scala.concurrent.Future[swagger.codegen.pet]])play.api.libs.json.JsValue.
Unspecified value parameter tjs.
      Ok(Json.toJson(deletePetImpl(id)))
                    ^
Error:(40, 21) No Json serializer found for type scala.concurrent.Future[Option[swagger.codegen.pet]]. Try to implement an implicit Writes or Format for this type.
      Ok(Json.toJson(findPetByIdImpl(id)))
                    ^
Error:(40, 21) not enough arguments for method toJson: (implicit tjs: play.api.libs.json.Writes[scala.concurrent.Future[Option[swagger.codegen.pet]]])play.api.libs.json.JsValue.
Unspecified value parameter tjs.
      Ok(Json.toJson(findPetByIdImpl(id)))
                    ^                    ^

**Comment by andreaTP** [comment]
Hi!
If you need to generate Play async you need to also uncomment this line:
https://github.com/unicredit/sbt-swagger-codegen-examples/blob/master/build.sbt#L13

and then regenerate the code with:

```
swaggerClean
swaggerCodeGen
swaggerPlayClientCodeGen
swaggerPlayServerCodeGen
```

**Comment by wontheone1** [comment]
@andreaTP , thanks again for the swift response. With the instructions, I was able to test the code generation and API. (One thing that wasn't explicitly mentioned was I had to refactor PetStoreControllerImpl.scala( in eu.unicredit package) file to same location as where Model.scala( in swagger.codegen package) is because, in my intelli J the model was not imported properly and giving me red colors. So I think it is better to be located with in the same package from the first in my opinion.) 

p.s. I found the logic of commented implementations were wrong, which I can elaborate in another issue.

**Comment by andreaTP** [comment]
I'm sorry for the integration with IntelliJ, but unfortunatelly IDE doesn't play well with source code generation ....
I double test everything with sbt and it works fine, could you double check that it is just an IDE fault?

p.s.
not sure on how to explain people to implement such a piece of code ... :-S I'm open to suggestions!

**Comment by wontheone1** [comment]
I think I had a problem because I didn't know I should have server and client as separate project and should not run both 'swaggerPlayClientCodeGen, swaggerPlayServerCodeGen' in one project. Also I am not sure its because of the update, but now import works normal and no red color. Thanks. And yeah, even though 'sbt' and 'run' works fine, but I see a lot of red colors once I import and try to build from intelliJ, so it doesn't go well with IDE may be. I am closing this now.

**Comment by fralken** [comment]
The example you tried before wasn't up to date, that's why the errors you encountered. Now that it is updated to the latest version, everything should work again.

And yes, I confirm IntelliJ has problems in coloring generated code (I see red colors too). But this is a fault of IntelliJ, not the code itself.
